editors/emacs/swarm-mode.el

```diff
@@ -7,151 +7,170 @@
-      (let* (
-             ;; Generate the current keywords with:
-             ;; cabal run swarm:swarm-docs -- editors --emacs
-             (x-keywords '("def" "tydef" "end" "let" "in" "require"))
-             (x-builtins '(
-               "self"
-               "parent"
-               "base"
-               "if"
-               "inl"
-               "inr"
-               "case"
-               "fst"
-               "snd"
-               "force"
-               "undefined"
-               "fail"
-               "not"
-               "format"
-               "chars"
-               "split"
-               "charat"
-               "tochar"
-               "key"
-             ))
-             (x-commands '(
-               "noop"
-               "wait"
-               "selfdestruct"
-               "move"
-               "backup"
-               "volume"
-               "path"
-               "push"
-               "stride"
-               "turn"
-               "grab"
-               "harvest"
-               "sow"
-               "ignite"
-               "place"
-               "ping"
-               "give"
-               "equip"
-               "unequip"
-               "make"
-               "has"
-               "equipped"
-               "count"
-               "drill"
-               "use"
-               "build"
-               "salvage"
-               "reprogram"
-               "say"
-               "listen"
-               "log"
-               "view"
-               "appear"
-               "create"
-               "halt"
-               "time"
-               "scout"
-               "whereami"
-               "waypoint"
-               "structure"
-               "floorplan"
-               "hastag"
-               "tagmembers"
-               "detect"
-               "resonate"
-               "density"
-               "sniff"
-               "chirp"
-               "watch"
-               "surveil"
-               "heading"
-               "blocked"
-               "scan"
-               "upload"
-               "ishere"
-               "isempty"
-               "meet"
-               "meetall"
-               "whoami"
-               "setname"
-               "random"
-               "run"
-               "return"
-               "try"
-               "swap"
-               "atomic"
-               "instant"
-               "installkeyhandler"
-               "teleport"
-               "as"
-               "robotnamed"
-               "robotnumbered"
-               "knows"
-               "east"
-               "north"
-               "west"
-               "south"
-               "down"
-               "forward"
-               "left"
-               "back"
-               "right"
-             ))
-             (x-types '("int" "text" "dir" "bool" "cmd" "void" "unit" "actor"))
+(defvar swarm-mode-types-regexp
+  (concat
+   "\\b"
+   (regexp-opt '("Int" "Text" "Dir" "Bool" "Cmd" "Void" "Unit" "Actor") t)
```

Comment:
How about just making this match on *any* word that starts with an uppercase letter?  If I use `tydef` to make my own type (which is required to start with a capital letter) it would be nice if my new type name is also highlighted like a type.

Technically it is possible to also create term-level variables that start with an uppercase letter, in which case those would be highlighted incorrectly, but in practice I don't think that's a big deal since I assume everyone will typically name their term variables starting with lowercase letters.